context_manager/instruction.md

```diff
@@ -0,0 +1,37 @@
+Context managers are a way of allocating and releasing some sort of resource exactly where you need it. The most widely used example of context managers is the 'with' statement.
+
+If you forget to close the file or let's say the database, it will later lead to big consequences. But with context manager, you don’t need to remember this, it will close everything for you.
+
+We can implement context manager as generator 
+
+```
+from contextlib import contextmanager
+@contextmanager
+```
+or as class by using magic functions
+```
+__init__
+__enter__
+__exit__
+```
+### LEVEL-1.
+
+1. Look at the file write_into_file.py
+
+2. Look at the tests in tests folter 
+
+3. Run tests with command:
+
+    ```python3 -m unittest tests/tests_for_level_one.py```
```

Comment:
test's filename should be exactly the same with the file you are testing.

in your case:
  test_write_into_file.py